_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -1405,6 +1422,26 @@ mfxStatus ImplementationAvc::Reset(mfxVideoParam *par)
 
     mfxExtCodingOption3 & extOpt3New = GetExtBufferRef(newPar);
     mfxExtCodingOption3 & extOpt3Old = GetExtBufferRef(m_video);
+
+
+    if (extOpt2New.IntRefCycleSize == 1)
+    {
+        if (extOpt2New.IntRefType == MFX_REFRESH_SLICE)
+        {
+            extOpt2New.IntRefCycleSize = extOpt3New.NumSliceP;
+        }
+        else
+        {
+            // set intra refresh cycle to 1 sec
+            extOpt2New.IntRefCycleSize =
+                (mfxU16)((newPar.mfx.FrameInfo.FrameRateExtN + newPar.mfx.FrameInfo.FrameRateExtD - 1) /
+                         newPar.mfx.FrameInfo.FrameRateExtD);
+        }
+        mfxExtCodingOption2 & extOpt2Par = GetExtBufferRef(*par);
+        extOpt2Par.IntRefCycleSize = extOpt2New.IntRefCycleSize;
+        checkStatus = MFX_WRN_VIDEO_PARAM_CHANGED;
+    }
+
```

Comment:
Same question as above

Reply:
For `ImplementationAvc::Reset` verification can't be moved inside `IntRefType` condition, because it is inside another condition which is true only when starting new sequence, but invalid value need to be processed regardless of that, so i added another check to condition.
The answer to another parts of your question is pretty much the same as above except that `CheckVideoParam` is called inside `ProcessAndCheckNewParameters` here 
https://github.com/Intel-Media-SDK/MediaSDK/blob/7e3ef73f9b4503f68721d9b329536bf3d119dddb/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp#L1264
And `ProcessAndCheckNewParameters` called at the beginning of `ImplementationAvc::Reset`